core/arxiv/submission/services/classic/__init__.py

```diff
@@ -0,0 +1,664 @@
+        except OperationalError as e:
+            raise Unavailable('Classic database unavailable') from e
+    return inner
+
+
+def is_available(**kwargs: Any) -> bool:
+    """Check our connection to the database."""
+    try:
+        current_session().query("1").from_statement(text("SELECT 1")).all()
+    except Exception as e:
+        logger.error('Encountered an error talking to database: %s', e)
+        return False
+    return True
+
+
+@retry(ClassicBaseException, tries=3, delay=1)
+@handle_operational_errors
+def get_licenses() -> List[License]:
+    """Get a list of :class:`.domain.License` instances available."""
+    license_data = current_session().query(models.License) \
+        .filter(models.License.active == '1')
+    return [License(uri=row.name, name=row.label) for row in license_data]
+
+
+@retry(ClassicBaseException, tries=3, delay=1)
+@handle_operational_errors
+def get_events(submission_id: int) -> List[Event]:
+    """
+    Load events from the classic database.
+
+    Parameters
+    ----------
+    submission_id : int
+
+    Returns
+    -------
+    list
+        Items are :class:`.Event` instances loaded from the class DB.
+
+    Raises
+    ------
+    :class:`.classic.exceptions.NoSuchSubmission`
+        Raised when there are no events for the provided submission ID.
+
+    """
+    session = current_session()
+    event_data = session.query(DBEvent) \
+        .filter(DBEvent.submission_id == submission_id) \
+        .order_by(DBEvent.created)
+    events = [datum.to_event() for datum in event_data]
+    if not events:      # No events, no dice.
+        raise NoSuchSubmission(f'Submission {submission_id} not found')
+    return events
+
+
+@retry(ClassicBaseException, tries=3, delay=1)
+@handle_operational_errors
+def get_user_submissions_fast(user_id: int) -> List[Submission]:
+    """
+    Get all active submissions for a user.
+
+    Uses the same approach as :func:`get_submission_fast`.
+
+    Parameters
+    ----------
+    submission_id : int
+
+    Returns
+    -------
+    list
+        Items are the user's :class:`.domain.submission.Submission` instances.
+
+    """
+    session = current_session()
+    db_submissions = list(
+        session.query(models.Submission)
+        .filter(models.Submission.submitter_id == user_id)
+        .order_by(models.Submission.doc_paper_id.desc())
+    )
+    grouped = groupby(db_submissions, key=lambda dbs: dbs.doc_paper_id)
+    submissions: List[Submission] = []
+    for arxiv_id, dbss in grouped:
+        logger.debug('Handle group for arXiv ID %s: %s', arxiv_id, dbss)
+        if arxiv_id is None:    # This is an unannounced submission.
+            for dbs in dbss:    # Each row represents a separate e-print.
+                submissions.append(load.to_submission(dbs))
+        else:
+            dbss = sorted(dbss, key=lambda dbs: dbs.submission_id)
+            submissions.append(load.load(dbss))
+    return [s for s in submissions if not s.deleted]
```

Comment:
```suggestion
    return [subm for subm in submissions if not subm.deleted]
```
since the rest of this function is trying to avoid single-character variables ;-)

Reply:
Fixed in ec8955b